src/Parser/FileParser.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Spatie\CodeOutline\Parser;
+
+use Spatie\CodeOutline\Elements\EmptyLine;
+use Spatie\CodeOutline\Elements\Line;
+use Spatie\CodeOutline\Elements\Page;
+
+class FileParser implements Parser
+{
+    /** @var string */
+    protected $path;
+
+    public function __construct(string $path)
+    {
+        $this->path = $path;
+    }
+
+    public function getParsed(): Page
+    {
+        $contents = file_get_contents($this->path);
```

Comment:
Does this file path need to check whether it's existed?
How about using the `file_exists` function to do this?

Reply:
I don't want this part of the code to concern itself with that kind of high level error handling. Whether the file exists or not should be done on a higher level. I've added the check when constructing the `FileParser`.

Reply:
Yes. The proper way is to check whether the file is existed in `FileParser` class constructor :+1: .